src/Symfony/Component/RateLimiter/CompoundLimiter.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+/**
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+class CompoundLimiter implements LimiterInterface
+{
+    private $limiters;
+
+    /**
+     * @param LimiterInterface[] $limiters
+     */
+    public function __construct(array $limiters)
+    {
+        $this->limiters = $limiters;
+    }
+
+    public function consume(int $tokens = 1): bool
+    {
+        $allow = true;
+        foreach ($this->limiters as $i => $limiter) {
+            $allow = $limiter->consume($tokens) && $allow;
```

Comment:
should this be: 

```suggestion
            $allow = $allow && $limiter->consume($tokens);
```

? ( short circuit )

Reply:
This changes the behavior of the compound limiter (as `consume()` is not called when `$allow` is already false). I'm not sure about the expected behavior with compound limiters, but counting all hits on all limiters is at least consistent (otherwise, some limiters will be called and others won't, unless we sort them with the most likely to reach its it limit as first).